test/model/database_test.dart

```diff
@@ -41,6 +41,60 @@ void main() {
         'acked_push_token': null,
       });
     });
+
+    test('create account with same realm and userId ', () async {
+      final accountData = AccountsCompanion.insert(
+        realmUrl: Uri.parse('https://chat.example/'),
+        userId: 1,
+        email: 'asdf@example.org',
+        apiKey: '1234',
+        zulipVersion: '6.0',
+        zulipMergeBase: const Value('6.0'),
+        zulipFeatureLevel: 42,
+      );
+      final accountDataWithSameUserId = AccountsCompanion.insert(
+        realmUrl: Uri.parse('https://chat.example/'),
+        userId: 1,
+        email: 'otheremail@example.org',
+        apiKey: '12345',
+        zulipVersion: '6.0',
+        zulipMergeBase: const Value('6.0'),
+        zulipFeatureLevel: 42,
+      );
+      await database.createAccount(accountData);
+      try  {
+        await database.createAccount(accountDataWithSameUserId);
+      } catch (e) {
+        check(e).isA<AccountAlreadyExistsException>();
+      }
```

Comment:
Handy `check` shortcut for this:
```suggestion
      await check(database.createAccount(accountDataWithSameUserId))
        .throws<AccountAlreadyExistsException>();
```